src/browsers/base/GeneResultsPage/GeneResultsUmapPlot.js

```diff
@@ -21,60 +23,123 @@ const DEFAULT_COLORS = [
-      type: 'scatter',
-      data: {
-        labels: new Set(labels),
-        datasets: chartData,
-      },
-      options: {
-        ...otherProps,
-        responsive: true,
-        plugins: {
-          legend: {
-            position: 'top',
-          },
-          title: {
-            display: true,
-            text: 'UMAP of residual expression across all genes',
-          },
-        },
-      },
-    })
-
-    setChart(newChart)
-  }, [embedding, labels, labelColors])
-
-  return <canvas id={id} />
+    const colourKeys = lodash.keys(datasets)
+
+    const Tooltip = select(wrapperRef.current).select('.tooltip')
+
+    Tooltip.style('opacity', 0)
+      .attr('class', 'tooltip')
+      .style('background-color', 'white')
+      .style('border', 'solid')
+      .style('border-width', '2px')
+      .style('border-radius', '5px')
+      .style('padding', '5px')
+      .style('position', 'absolute')
+
+    svgContent
+      .selectAll('.mydot')
+      .data(embedding)
+      .join('circle')
+      .attr('class', 'mydot')
+      .attr('r', 5)
+      .attr('cx', (d) => xScale(d[0]))
+      .attr('cy', (d) => yScale(d[1]))
+      .attr('fill-opacity', (d, i) => (labels[i] === hover ? 1 : 0.3))
+      .attr('fill', (d, i) => labelColors[colourKeys.indexOf(labels[i])])
+      .on('mouseover', (e, d) => {
+        setHover(labels[embedding.indexOf(d)])
+        Tooltip.style('opacity', 1)
+      })
+      .on('mouseleave', () => {
+        setHover(null)
+        Tooltip.style('opacity', 0)
+      })
+      .on('mousemove', (e, d) => {
+        Tooltip.html(labels[embedding.indexOf(d)])
+          .style('opacity', 1)
+          .style('left', `${xScale(d[0])}px`)
+          .style('top', `${yScale(d[1]) + 10}px`)
+      })
+
+    const xAxis = axisBottom(xScale)
+    svg
+      .select('.x-axis')
+      .attr('transform', `translate(0, ${height - 30})`)
+      .call(xAxis)
+
+    const yAxis = axisLeft(yScale)
+    svg.select('.y-axis').attr('transform', `translate(30, 0)`).call(yAxis)
+
+    // zoom
+    const zoomBehaviour = zoom()
+      .scaleExtent([0.5, 5])
+      .translateExtent([
+        [0, 0],
+        [width, height],
+      ])
+      .on('zoom', (e) => {
+        setCurrentZoomState(e.transform)
+      })
+    svg.call(zoomBehaviour)
+  }, [currentZoomState, embedding, labels, hover])
+
+  return (
+    <>
+      <div ref={wrapperRef}>
+        <svg ref={svgRef} height={height} width={width} style={{ border: '1px solid black' }}>
+          <defs>
+            <clipPath id={id}>
+              <rect x="0" y="0" width="100%" height="100%" />
+            </clipPath>
+          </defs>
+          <g className="content" clipPath={`url(#${id})`} />
+          <g className="x-axis" />
+          <g className="y-axis" />
+        </svg>
+        <div className="tooltip" />
+      </div>
+    </>
+  )
 }
 
 GeneResultsUmapPlot.propTypes = {
+  height: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired,
```

Comment:
Set defaults for height and width instead of being required

```suggestion
  height: PropTypes.number,
  width: PropTypes.number,
```